test/testcafe/framework/shared/index.js

```diff
@@ -5,9 +5,22 @@ import { t } from 'testcafe';
 // 2. Widget transitions to a new page and animations have finished (afterRender)
 // 3. Context object from the 'afterReady' event 
 export async function checkConsoleMessages(context = {}) {
+  if (!Array.isArray(context)) {
+    context = ['ready', 'afterRender', context];
+  }
+
   const { log } = await t.getBrowserConsoleMessages();
-  await t.expect(log.length).eql(3);
-  await t.expect(log[0]).eql('===== playground widget ready event received =====');
-  await t.expect(log[1]).eql('===== playground widget afterRender event received =====');
-  await t.expect(JSON.parse(log[2])).eql(context);
+  await t.expect(log.length).eql(context.length);
+  for (let i = 0; i < context.length; i++) {
+    switch (context[i]) {
```

Comment:
With this change, we're no longer expecting the log statements to appear in a specific order. I think that this is OK for most cases, but we may want to provide an additional `checkConsoleMessagesStrict` in the future to validate the ordering is adhered to.

Reply:
I think I'm reading the code a little differently; I see it very strictly checking every log line that it matches what is passed in context array. Previous behavior is preserved by converting context object to ['ready', 'afterRender', context]. This will check that those 3 expected messages occur in the log in that sequence. Repeated renderings or view changes result in the sequence being lengthened. A test can assert that the full sequence happens exactly as expected: https://github.com/okta/okta-signin-widget/pull/1653/files#diff-adda49d89411ec7c49ad608f0b9a3a8333057b36448d5c8e3929229a5d51b52cR137